### PR TITLE
Solve #1438 MySQL errors showing up in search submission

### PR DESCRIPTION
**Pull request by Ngai-E**
@aethelwulffe @muarachmann  I succeeded to make this fix kindly review and give me feedback.
Screenshot from 2019-03-15 12-26-30
After inputting selecting desired drop down menu options and clicking submit, results are shown as needed

### CODE REVIEW

File: interface/reports/reports_controllers/SalesByItemController.php [hunk 1]
```diff
@@ -262,21 +262,21 @@ function prepareQuery() {
...
     "fe.date >= '$from_date 00:00:00' AND fe.date <= '$to_date 23:59:59' " .
-    //"fe.date >= ? AND fe.date <= ? " .
     "WHERE s.fee != 0";
     array_push($sqlBindArray,$from_date,$to_date);
   // If a facility was specified.
   if ($form_facility) {
-    $query .= " AND fe.facility_id = ?";
-     array_push($sqlBindArray,$form_facility);
+    $query .= " AND fe.facility_id = $form_facility";
+     //array_push($sqlBindArray,$form_facility);
```
Comment (aethelwulffe) [inline interface/reports/reports_controllers/SalesByItemController.php]: drop old code.

**Comment by aethelwulffe** [comment]
@muarachmann You see any issues here?  Test is good, Just trying to figure out what the heck happened to the array bind that presumably worked at one time (or did it?)

**Comment by Ngai-E** [comment]
@aethelwulffe  changes done!

**Comment by KoniKodes** [comment]
Hi, is it possible to have a new reviewed or merged?
This is the last week of the application round and I'm trying to get all the ducks in a row...

**Comment by Ngai-E** [comment]
@muarachmann @aethelwulffe
Made the changes you requested kindly review. I have been waiting

**Comment by muarachmann** [comment]
> @muarachmann You see any issues here? Test is good, Just trying to figure out what the heck happened to the array bind that presumably worked at one time (or did it?)

I too wondering. Just worried about sql injection. But i think it is a good fix for the errors

**Comment by Ngai-E** [comment]
@muarachmann will the array bind prevent the mysql injection? If so, i think u better guide me how i can go about implementing it. I would love to improve the code if possible except you are ok with the solution so far

**Comment by muarachmann** [comment]
Yes the array bind will replying from phone. You could try by doing a dump
and die before the query runs to see the output. I am sure the array is
empty reason for that error. Just try and see how to better parse it

**Comment by Ngai-E** [comment]
@muarachmann i did print the content of the array and noticed it was empty. You are right that was the reason for the error

**Comment by muarachmann** [comment]
Okay... So ensure you are getting it correctly then the bind should work

**Pull request merged**